### PR TITLE
Refactor CmdArgs

### PR DESCRIPTION
**Pull request by singloon**
Closes https://github.com/Verisimilitude11/DNAnalyzer/issues/189

### CODE REVIEW

**Comment by Verisimilitude11** [comment]
@singloon could you please resolve the conflicts?

**Comment by singloon** [comment]
@Verisimilitude11 resolved the conflicts

**Comment by singloon** [comment]
@Verisimilitude11 I think everything is resolved now for code review now @Verisimilitude11

**Comment by singloon** [comment]
@Verisimilitude11 please can you review, or else I'll have to rebase every-time there are changes.

**Comment by singloon** [comment]
I don't want to resolve conflicts again @Verisimilitude11

**Comment by Verisimilitude11** [comment]
Yes, sorry about that, this PR is first in my backlog review.

**Comment by Verisimilitude11** [comment]
Hey @singloon, I will be out of office for some time, so it will take me some time merge the pull request. However, I will approve the pull request in case you are participating in Hacktoberfest. Thanks for working on this, we at DNAnalyzer really appreciate your help!

**Review by Verisimilitude11 — APPROVED** [review]

**Comment by Verisimilitude11** [comment]
@singloon could you please clear the merge conflicts one last time? Thanks, I really appreciate it!

**Comment by singloon** [comment]
@Verisimilitude11 resolved conflicts. Please can you review and merge accordingly, I see other PRs that are newer being merged. Thus causing the conflicts.

File: src/main/java/DNAnalyzer/CmdArgs.java [hunk 1]
```diff
@@ -74,85 +76,36 @@ public class CmdArgs implements Runnable {
...
-      // Find the longest protein in DNA
-      ProteinAnalysis.printLongestProtein(proteins);
+    dnaAnalyzer
+      .printProteins()
+      .outPutCodons(minCount, maxCount)
+      .printLongestProtein();
+  }
 
-      // Print if DNA is random
-      if (Properties.isRandomDNA(dna)) {
```
Comment (Verisimilitude11) [inline src/main/java/DNAnalyzer/CmdArgs.java:101]: @singloon it seems like you accidentally removed the call to isRandomDNA(). Could you please add this back?

**Review by Verisimilitude11 — CHANGES_REQUESTED** [review]

File: src/main/java/DNAnalyzer/CmdArgs.java (hunk 1 above)
Comment (singloon) [reply src/main/java/DNAnalyzer/CmdArgs.java:101]: added it back

Comment (Verisimilitude11) [reply src/main/java/DNAnalyzer/CmdArgs.java:101]: cool

Comment (Verisimilitude11) [reply src/main/java/DNAnalyzer/CmdArgs.java:101]: Thanks for adding that back! It also seems that calling the nucleotide function returns an error - which is not present in main:
image
Could you please look into that as well! Thank you so much for all of your help!!

Comment (singloon) [reply src/main/java/DNAnalyzer/CmdArgs.java:101]: What's your input, the printNucleotideCount function hasn't changed.

When I test it seems fine
`java -jar build/libs/DNAnalyzer.jar assets/dna/random/dnalong.fa --amino=ser --min=10 --max=100
Proteins coded for Serine:
----------------------------------------------------
1. TCTCATGATCACCAGTTCTGACGTTACAGTATTTTCGGTTGAGCAGGCCCCATGGGGCCCCCGCATGCCGAATTACGATATGATGCCCACTATCCTGTGTCTTCCAACCTTATGACTGACTTGTATGCGCTGCGAGGTCCCTCGATAGATTTGCTCCCACCCGTCCCGGAAACCATATCGACGACTTGTAGGTCTCTAA
2. TCCTGTGTCTTCCAACCTTATGACTGACTTGTATGCGCTGCGAGGTCCCTCGATAGATTTGCTCCCACCCGTCCCGGAAACCATATCGACGACTTGTAGGTCTCTAA
3. TCAAAACCAATCTCATGATCACCAGTTCTGACGTTACAGTATTTTCGGTTGAGCAGGCCCCATGGGGCCCCCGCATGCCGAATTACGATATGATGCCCACTATCCTGTGTCTTCCAACCTTATGACTGACTTGTATGCGCTGCGAGGTCCCTCGATAGATTTGCTCCCACCCGTCCCGGAAACCATATCGACGACTTGTAGGTCTCTAA
4. TCGGTTGAGCAGGCCCCATGGGGCCCCCGCATGCCGAATTACGATATGATGCCCACTATCCTGTGTCTTCCAACCTTATGACTGACTTGTATGCGCTGCGAGGTCCCTCGATAGATTTGCTCCCACCCGTCCCGGAAACCATATCGACGACTTGTAGGTCTCTAA
5. AGTTCTGACGTTACAGTATTTTCGGTTGAGCAGGCCCCATGGGGCCCCCGCATGCCGAATTACGATATGATGCCCACTATCCTGTGTCTTCCAACCTTATGACTGACTTGTATGCGCTGCGAGGTCCCTCGATAGATTTGCTCCCACCCGTCCCGGAAACCATATCGACGACTTGTAGGTCTCTAA
6. AGCAGGCCCCATGGGGCCCCCGCATGCCGAATTACGATATGATGCCCACTATCCTGTGTCTTCCAACCTTATGACTGACTTGTATGCGCTGCGAGGTCCCTCGATAGATTTGCTCCCACCCGTCCCGGAAACCATATCGACGACTTGTAGGTCTCTAA

GC-content (genome): 0.16777216

Nucleotide count:
A: 25000381 (25.000381%)
T: 24998528 (24.998528%)
G: 25000967 (25.000969%)
C: 25000124 (25.000126%)

Codons in reading frame 1 (10-100 occurrences):
----------------------------------------------------

Longest gene (209 nucleotides): TCAAAACCAATCTCATGATCACCAGTTCTGACGTTACAGTATTTTCGGTTGAGCAGGCCCCATGGGGCCCCCGCATGCCGAATTACGATATGATGCCCACTATCCTGTGTCTTCCAACCTTATGACTGACTTGTATGCGCTGCGAGGTCCCTCGATAGATTTGCTCCCACCCGTCCCGGAAACCATATCGACGACTTGTAGGTCTCTAA

dnalong.fa has been detected to be random.`

Comment (Verisimilitude11) [reply src/main/java/DNAnalyzer/CmdArgs.java:101]: Huh, weird, it works now. I'm going to merge this PR.

**Review by Verisimilitude11 — APPROVED** [review]

**Comment by singloon** [comment]
Thanks @Verisimilitude11

**Comment by Verisimilitude11** [comment]
> Thanks @Verisimilitude11 

No problem! Sorry for asking you to resolve conflicts so many times.

**Pull request merged**